packages/http-client-java/generator/http-client-generator/src/main/java/com/microsoft/typespec/http/client/generator/TypeSpecPlugin.java

```diff
@@ -44,11 +45,17 @@ public class TypeSpecPlugin extends Javagen {
 
     private final EmitterOptions emitterOptions;
 
-    private final Map<String, String> crossLanguageDefinitionsMap = new TreeMap<>();
+    private TypeSpecMetadata metadata;
 
     public Client processClient(CodeModel codeModel) {
+        this.metadata = new TypeSpecMetadata(ClientModelUtil.getArtifactId(), emitterOptions.getFlavor(),
+            emitterOptions.getApiVersion() == null
+                ? MetadataUtil.getLatestApiVersionFromClient(codeModel)
+                : emitterOptions.getApiVersion());
```

Comment:
For my learning; where do we set the "crossLanguageDefinitions" on a "TypeSpecMetadata" instance? I I might have overlooked but couldn't find it.